numpy/_core/tests/test_ufunc.py

```diff
@@ -1668,20 +1668,28 @@ def test_scalar_reduction(self):
         # Check scalar behaviour for ufuncs without an identity
         assert_equal(np.power.reduce(3), 3)
 
-        # Make sure that scalars are coming out from this operation
-        assert_(type(np.prod(np.float32(2.5), axis=0)) is np.float32)
-        assert_(type(np.sum(np.float32(2.5), axis=0)) is np.float32)
-        assert_(type(np.max(np.float32(2.5), axis=0)) is np.float32)
-        assert_(type(np.min(np.float32(2.5), axis=0)) is np.float32)
-
-        # check if scalars/0-d arrays get cast
-        assert_(type(np.any(0, axis=0)) is np.bool)
+        # If axis=None, scalars should be produced
+        assert type(np.prod(np.float32(2.5), axis=None)) is np.float32
+        assert type(np.sum(np.float32(2.5), axis=None)) is np.float32
+        assert type(np.max(np.float32(2.5), axis=None)) is np.float32
+        assert type(np.min(np.float32(2.5), axis=None)) is np.float32
+        # TODO: In a sense should return an array, but this is axis=0 is
```

Comment:
Agree, arguably `axis=0` on a 0-D array should just fail! 

More relevantly here, are you sure you want to insist this returns a scalar? I guess just from being an operation on scalars only?